cloud-aws-common/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/common/validator/AwsLogRolePermissionValidator.java

```diff
@@ -35,21 +35,38 @@ public class AwsLogRolePermissionValidator extends AbstractAwsSimulatePolicyVali
     @Inject
     private LocationHelper locationHelper;
 
+    public void validateLog(AmazonIdentityManagementClient iam, InstanceProfile instanceProfile,
+            CloudS3View cloudFileSystem, String logsLocationBase, ValidationResultBuilder validationResultBuilder) {
+        Arn instanceProfileArn = Arn.of(instanceProfile.getArn());
+        Map<String, String> logReplacements = Map.ofEntries(
+                Map.entry("${ARN_PARTITION}", instanceProfileArn.getPartition()),
+                Map.entry("${LOGS_LOCATION_BASE}", removeProtocol(logsLocationBase)),
+                Map.entry("${LOGS_BUCKET}", locationHelper.parseS3BucketName(logsLocationBase))
+        );
+        Policy logPolicy = awsIamService.getPolicy("aws-cdp-log-policy.json", logReplacements);
+        validate(iam, instanceProfile, cloudFileSystem, logsLocationBase, validationResultBuilder, logPolicy);
+    }
+
+    public void
+    validateBackup(AmazonIdentityManagementClient iam, InstanceProfile instanceProfile,
+            CloudS3View cloudFileSystem, String backupLocationBase, ValidationResultBuilder validationResultBuilder) {
+        Arn instanceProfileArn = Arn.of(instanceProfile.getArn());
+        Map<String, String> backupReplacements = Map.ofEntries(
```

Comment:
Instead of copying it many times could you please create a util or factory class for this Map initialisation?

Reply:
MAP is initialized with different values in the different cases.

Reply:
I meant this is the same like in case of:
https://github.com/hortonworks/cloudbreak/pull/12882/files#diff-9c81e48d8a883c9bbd7cfa6b1983c8e570575081b6bdbedddc9992887d2e7df2R205-R209

Reply:
I am still seeing the duplication of the replacement Map build logic, it builds a map with the same keyset. Can we have a builder for it and not to build it with duplicated logic?